packages/nimble-components/src/combobox/specs/IxD.md

```diff
@@ -0,0 +1,241 @@
+| Menu        | A list of options to choose from displayed as an _open_ state           |
+| Menu item   | A selection the end-user can make, shown with other options in the menu |
+
+#### Label
+
+Comboboxes should always have a label, unless approved by an interaction designer or user researcher (usually in toolbar use cases). _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Field Input
+
+Comboboxes can be empty by default or have a default selection from the menu. Empty comboboxes should include placeholder text that propts the end-user to enter a custom value or select an option from the menu. Placeholder text should always follow the pattern "Enter or select [thing(s)]", for example "Enter or select address". Ellipses are not needed. Use sentence casing.
+
+![ ](spec-images/combobox-2.png)
+
+#### Menu & Menu Items
+
+A list of options to select, displayed as an open state._See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+### Related Components
+
+To allow end-users to select a menu item but not enter a custom value, use a [**select**](/packages/nimble-components/src/select/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-select.png)
+
+To allow end-users to multi-select menu items, use a [**tag picker**](/packages/nimble-components/src/tag-picker/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-picker.png)
+
+## Design
+
+### Configuration
+
+#### Label
+
+The client-user should be able to specify where they want the label to be positioned. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Icons
+
+The client-user should be able to specify whether they want left icons on menu items or not. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Groups
+
+The client-user should be able to create groups for menu items. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client should be able to specify whether they want the group name to be visible on the component when the menu is closed.
+
+![ ](spec-images/combobox-26.png)
+
+#### Secondary Text
+
+The client-user should be able to specify whether they want secondary text on menu items or not. _See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information._
+
+#### Prefix & Suffix
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify a prefix and/or suffix that appears in the field.
+
+![ ](spec-images/combobox-3.png)
+
+#### Autocomplete
+
+The client-user should be able to specify autocomplete behavior (inline, list, both, none). Autocomplete is accessed via the field input (not pictured). _See [current implementation](https://nimble.ni.dev/storybook/?path=/docs/components-combobox--docs) for more information._
```

Comment:
We actually have an opportunity here to break from the FAST behavior of how autocomplete works, and offer something that I think is more in demand, which is the ability to filter against _any_ text within the option text, instead of just from the front. The only reason, that I know of, for why FAST limits the filter to the front of the text is because of how they do inline autocomplete, where they highlight the rest of the text that has been autocompleted, which is all done within the input section of the component.

So, I mention this in case you wanted to conceive of your ideal behavior for how autocomplete should work for the `Combobox`. The main interaction that I think we need to describe in detail, that is an unknown at the moment, is if the `Combobox` has an `autocomplete` of either `both` or `inline` (this results in the text in the input autocompleting, with untyped text currently being highlighted), what should happen as the user types, particularly if the text matches text _not_ at the front of the option text. The main alternative I can think of is that we don't highlight any of the autocompleted text, but we still update the text in the input field. The drawback to this is that your cursor could be jumping around if your match was triggering against the beginning at times and other places as you continue typing.